openshift-storage-libs/openshiftstoragelibs/baseclass.py

```diff
@@ -246,25 +268,35 @@ def _create_storage_class(self, provisioner, secret_name=None,
                               hacount=None,
                               is_arbiter_vol=False, arbiter_avg_file_size=None,
                               heketi_zone_checking=None, volumeoptions=None,
-                              skip_cleanup=False):
+                              skip_cleanup=False, glusterfs_registry=False):
 
         # Create secret if one is not specified
         if not secret_name:
-            secret_name = self.create_secret(skip_cleanup=skip_cleanup)
+            secret_name = self.create_secret(
+                skip_cleanup=skip_cleanup, glusterfs_registry=glusterfs_registry)
 
         # Create storage class
         secret_name_option = "secretname"
         secret_namespace_option = "secretnamespace"
         if provisioner != "kubernetes.io/glusterfs":
             secret_name_option = "rest%s" % secret_name_option
             secret_namespace_option = "rest%s" % secret_namespace_option
-        parameters = {
-            "resturl": self.sc["resturl"],
-            "restuser": self.sc["restuser"],
-            secret_name_option: secret_name,
-            secret_namespace_option: self.sc.get(
-                "secretnamespace", self.sc.get("restsecretnamespace")),
-        }
+        if glusterfs_registry:
+            parameters = {
+                "resturl": self.registry_sc["resturl"],
```

Comment:
Keep same indentation in same function. Use `dict.get` in all places in the function to get value.

Reply:
Done